src/test/java/stepDefinitions/SEDTweaksSteps.java

```diff
@@ -12,7 +26,171 @@ public SEDTweaksSteps()
+		expanded.click();
+					
+		WebElement expanded1 = driver.findElement(By.cssSelector("div.panel:nth-child(4) > div:nth-child(1) > a:nth-child(1)"));
+		expanded1.click();
+		
+		if(driver.getPageSource().contains("No Certification Criteria were tested for SED."))
+		    System.out.println("No Certification Criteria were tested for SED.");
+		else
+			driver.getPageSource().contains("Description of Intended Users");
+	}
+
+	@Then("^the header should be named Description of Intended Users$")
+	public void the_header_should_be_named_Description_of_Intended_Users() throws Throwable {
+	    
+		assertTrue(
+				driver.getPageSource().contains("Description of Intended Users"));
+			
+	}
+
+	@When("^I look at the 'SED Testing Completion Date'$")
+	public void i_look_at_the_SED_Testing_Completion_Date() throws Throwable {
+		if(driver.getPageSource().contains("No Certification Criteria were tested for SED."))
+		    System.out.println("No Certification Criteria were tested for SED.");
+		else
+			driver.getPageSource().contains("Date SED Testing was Completed");
 	
+	}
+
+	@Then("^the header should be named 'Date SED Testing was Completed'$")
+	public void the_header_should_be_named_Date_SED_Testing_was_Completed() throws Throwable {
+	    assertTrue(
+				driver.getPageSource().contains("Date SED Testing was Completed"));
+	}
+
+	@When("^I look at the download button$")
+	public void i_look_at_the_download_button() throws Throwable {
+	   driver.findElement(By.cssSelector(".panel-body > div:nth-child(1) > span:nth-child(1) > span:nth-child(5) > table:nth-child(2) > tfoot:nth-child(2) > tr:nth-child(1) > td:nth-child(1) > button:nth-child(1)")).getTagName();
+	
+	}
+
+	@Then("^the button should be named 'Download SED Details'$")
+	public void the_button_should_be_named_Download_SED_Details() throws Throwable {
+		assertTrue(
+				driver.getPageSource().contains("Download SED Details"));
+	}
+
+	@When("^I look at the associated criteria$")
+	public void i_look_at_the_associated_criteria() throws Throwable {
+		driver.get("https://chpl.ahrqstg.org/#/product/8481");
+		WebDriverWait wait = new WebDriverWait(driver, 20);
+		wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("div.panel:nth-child(4) > div:nth-child(1) > a:nth-child(1)")));
+		
+	}
+
+	@Then("^the criteria number should also include the criteria title$")
+	public void the_criteria_number_should_also_include_the_criteria_title() throws Throwable {
+		Thread.sleep(5000);
+		WebElement expanded = driver.findElement(By.cssSelector("div.panel:nth-child(2) > div:nth-child(1) > a:nth-child(1)"));
+		expanded.click();
+					
+		WebElement expanded1 = driver.findElement(By.cssSelector("div.panel:nth-child(4) > div:nth-child(1) > a:nth-child(1)"));
+		expanded1.click();
+		
+		 WebElement Table_1 = driver.findElement(By.xpath("//*[@id='mainContent']/div[2]/div[2]/div/div/div[3]/div[2]/div/span/span[4]/table/thead/tr/th[1]"));
```

Comment:
Can/should we start putting IDs on tables? Or something? This xpath is way too deep, and that means it's very fragile. A single added (or removed) span anywhere in here and the test breaks.

Reply:
Probably not necessary to change for the PR. As AQA goes along, Ashwini should make a note to ask for IDs anytime she has to write expressions like that